File: DsaWithCpp/LinkedList/Node.cpp
struct Node
{
    int data;
    Node *next;
    Node *prev;

    Node()
    {
    }

    Node(int val)
    {
        data = val;
        next = prev = NULL;
    }

    Node(int val, Node *leftptr, Node *rightptr)
    {
        data = val;
        next = leftptr;
        prev = rightptr;
    }
};

File: Java/linkedlist/Node.java
package linkedlist;

class Node {
    int data;
    Node next;
    Node prev;

    Node() {

    }

    Node(int data) {
        this.data = data;
        this.next = this.prev = null;
    }

    Node(int data, Node next, Node prev) {
        this.data = data;
        this.next = next;
        this.prev = prev;
    }
}
